utils/fixed_decimal/src/decimal.rs

```diff
@@ -495,6 +495,145 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    fn n_magnitude(&self) -> i16 {
+        if self.digits.len() == 0 {
+            return self.magnitude;
+        }
+
+        self.magnitude - (self.digits.len() as i16 - 1)
+    }
+
+    fn increment_abs_by_one(&mut self) {
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                return;
+            }
+            self.digits[i] = 0;
+        }
+
+        // Still there is a carry, add `1` in the beginning of the digits.
+        self.digits.insert(0, 1);
+        self.magnitude = {
+            if self.digits.len() == 1 {
+                self.magnitude
+            } else {
+                self.magnitude + 1
+            }
+        };
+
+        if self.upper_magnitude < self.magnitude {
+            self.upper_magnitude = self.magnitude;
+        }
+
+        // remove trailing zeros from `digits`
+        for i in (0..self.digits.len()).rev() {
+            if self.digits[i] == 0 {
+                self.digits.pop();
+            } else {
+                break;
+            }
+        }
+    }
+
+    /// NOTE: accept only n <= magnitude
+    fn truncate(&mut self, n: i16) {
+        fn diff(x: i16, y: i16) -> u16 {
+            if x > y {
+                (x - y) as u16
```

Comment:
Issue: Code like this may overflow. For example, `x = 20000` and `y=-20000` would overflow since you are operating on `i16` values.

Use [i16::abs_diff](https://doc.rust-lang.org/std/primitive.i16.html#method.abs_diff) instead.

Reply:
Instead of `abs_diff`, use the functions I'm adding in https://github.com/unicode-org/icu4x/pull/1924

Reply:
Done